### ISSUE TITLE
Wrong type def in ApplicationCommandOptionChoiceData.value

### ISSUE THREAD

**Issue by kubahasek** [discordjs/guide#1233]
### Describe the bug

When I was writing my slash command today, I attempted to enter a value of a choice of 1, but then got an API error saying values can only be strings. The library allows to put numbers as well

### To Reproduce

1. Create a slash command
2. Add a choice where the value is a number.

### Device (please complete the following information)

macOS Monterrey
DiscordJS v14

**Comment by Jiralite** [comment]
Where in the guide did you receive this error from?

**Comment by kubahasek** [comment]
I only follow the guide for parts of it, as I am doing typescript my code looks a little different. This is the code it came from.
![Screenshot 2022-08-20 at 14 39 21](https://user-images.githubusercontent.com/28836407/185746592-5880946c-1891-40c3-8242-27ec11f4a75c.png)
t

**Comment by Jiralite** [comment]
So are you saying that payload when sent to Discord throws an API error?

**Comment by kubahasek** [comment]
That's correct. This error to be precise.
![Screenshot 2022-08-20 at 14 43 40](https://user-images.githubusercontent.com/28836407/185746731-8e0a051c-d967-46e1-a82c-0f86f6fd5779.png)

**Comment by Jiralite** [comment]
Why is that a bug? Your option is a string type?

**Comment by kubahasek** [comment]
Correct, but this is about the values. So basically what the code receives. Typescript allowed to put a number in there which resulted in this error.
